Automatically close PRs against staging repos

cblecker opened the pull request:
This adds a new commenter job to close PRs against staging repos

fixes https://github.com/kubernetes/kubernetes/issues/122238

cblecker commented:
/hold
for review and discussion

MadhavJivrajani commented on config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml:
```diff
@@ -131,6 +131,84 @@ periodics:
+
+        Please send feedback to sig-contributor-experience at [kubernetes/community](https://github.com/kubernetes/community).
+
+        /close not-planned
```
We can just use `/close` here, there is no special `not_planned` status for PRs unfortunately.
xref: https://github.com/kubernetes/test-infra/pull/27545

MadhavJivrajani commented on config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml:
````diff
@@ -131,6 +131,84 @@ periodics:
+        - The [staging area README](https://git.k8s.io/kubernetes/staging#readme), which describes how the staging repos function and how to contribute
+
+        Kubernetes Contributors:
+        If this PR is to update the `README.md` itself, please reopen this PR by commenting:
+        ```
+        /triage accepted
+        /reopen
+        ```
````
I'm likely missing something - can you pls point me to where we document that a PR to update the README is allowed for staging repos?

cblecker replied:
It's permitted in client-go, but not the others

MadhavJivrajani commented on config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml:
````diff
@@ -131,6 +131,84 @@ periodics:
+        Kubernetes Contributors:
+        If this PR is to update the `README.md` itself, please reopen this PR by commenting:
+        ```
+        /triage accepted
````
A consideration to keep in mind here would be `/triage accepted` can only be applied by org members iirc, if the contributor isn't an org member, they won't be able to re-open the PR.

MadhavJivrajani reviewed (commented):
Thanks for starting this!
Leaving a few initial comments <3

cblecker commented:
I dug a bit more and found https://github.com/kubernetes/test-infra/pull/22879 -- we already configured a blockade but we never turned the plugin on in client-go, so this change never took effect.

I've updated the blockage configuration to account for the fact that you can only change README.md in client-go, but not the other repos. I've also updated the closer job to key off of the blockade label.

BenTheElder commented on config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml:
```diff
@@ -131,6 +131,77 @@ periodics:
+      - --token=/etc/github-token/token
+      - --endpoint=http://ghproxy.default.svc.cluster.local
+      - |-
+        --comment=The Kubernetes project doesn't accept direct changes directly against this repository as outlined above.
```
It feels kinda weird to duplicate this text in the blockade comment AND the robo-/close comment : thinking :

jsoref replied:
"Closing per above"? (if you're confident the message will always be present)

jsoref commented:
@cblecker will you continue pushing this forward?
